model/project_parser.go

```diff
@@ -597,15 +596,21 @@ type GetProjectOpts struct {
 }
 
 func retrieveFile(ctx context.Context, opts GetProjectOpts) ([]byte, error) {
-	conf, err := evergreen.GetConfig()
-	if err != nil {
-		return nil, errors.Wrap(err, "can't get evergreen configuration")
+	if opts.Token == "" {
+		conf, err := evergreen.GetConfig()
+		if err != nil {
+			return nil, errors.Wrap(err, "can't get evergreen configuration")
+		}
+		ghToken, err := conf.GetGithubOauthToken()
+		if err != nil {
+			return nil, errors.Wrap(err, "can't get Github OAuth token from configuration")
+		}
+		opts.Token = ghToken
 	}
-	ghToken, err := conf.GetGithubOauthToken()
-	if err != nil {
-		return nil, errors.Wrap(err, "can't get Github OAuth token from configuration")
+	if opts.Revision == "" && opts.Ref != nil {
+		opts.Revision = opts.Ref.RemotePath
```

Comment:
This actually isn't quite right; Revision indicates the specific git hash that we want to look up the file from; when Revision is empty we default to HEAD which is typically right. opts.Ref.RemotePath is actually the file path name. (I'm not sure why you added this though; did it fix something..?) You may instead want to pass in opts.Ref.Branch (to specify the branch we want to look up the file from)

Reply:
that was supposed to be remotepath